### PR TITLE
Add project setting for max lights and reflections in GLES3

### PR DESCRIPTION
**Pull request by clayjohn**
Implemented based on https://github.com/godotengine/godot/pull/14902

Should alleviate https://github.com/godotengine/godot/issues/21957 for GLES3.

### CODE REVIEW

File: drivers/gles3/rasterizer_scene_gles3.cpp [hunk 1]
```diff
@@ -5064,6 +5064,10 @@ void RasterizerSceneGLES3::initialize() {
 
 	render_list.max_elements = GLOBAL_DEF_RST("rendering/limits/rendering/max_renderable_elements", (int)RenderList::DEFAULT_MAX_ELEMENTS);
 	ProjectSettings::get_singleton()->set_custom_property_info("rendering/limits/rendering/max_renderable_elements", PropertyInfo(Variant::INT, "rendering/limits/rendering/max_renderable_elements", PROPERTY_HINT_RANGE, "1024,1000000,1"));
+	render_list.max_lights = GLOBAL_DEF("rendering/limits/rendering/max_lights", (int)RenderList::DEFAULT_MAX_LIGHTS);
```
Comment (Calinou) [inline drivers/gles3/rasterizer_scene_gles3.cpp]: The setting name doesn't match the ones below (`"rendering/limits/rendering/max_renderable_lights"`), is this intended?

Comment (clayjohn) [reply drivers/gles3/rasterizer_scene_gles3.cpp]: Nope. :) copy-paste error.

**Comment by akien-mga** [comment]
Thanks!

**Comment by jknightdoeswork** [comment]
Thank you @clayjohn.

**Comment by akien-mga** [comment]
Cherry-picked for 3.1.2.

**Pull request merged**